src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -431,50 +440,44 @@ export class BundleDataClient {
       }, toBN(0));
   }
 
-  private async getLatestProposedBundleData(): Promise<{ bundleData: LoadDataReturnValue; blockRanges: number[][] }> {
+  async getPendingPoolRebalanceLeavesFromArweave(): Promise<PoolRebalanceRoot | undefined> {
```

Comment:
I've changed this function to only query once from Arweave on the pending bundle data if it exists. Otherwise return undefined. The relayer's inventory client already handles this fallback well https://github.com/across-protocol/relayer/blob/7863335cce2ecf36e917ab7f81fce11e3a31f9cc/src/clients/InventoryClient.ts#L644

and falls back to using the ExecutedRootBundle events in the HubPoolClient